autoload/ale/events.vim

```diff
@@ -92,6 +92,11 @@ function! ale#events#FileChangedEvent(buffer) abort
     endif
 endfunction
 
+function! ale#events#FocusGained(buffer) abort
+    call setbufvar(a:buffer, 'force_changedtick', 1)
+    call ale#Queue(0, 'lint_file', a:buffer)
```

Comment:
Check `g:ale_lint_on_focus_gained` here again, and probably `ale_enabled` like we do above. Otherwise it will be difficult to turn this event off after the `autocmd` command has been set.